dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemToReg.java

```diff
@@ -313,6 +314,15 @@ private void publishRegisters(Set<Register> registers) {
             }
         }
 
+        private AddressOffset toAddressOffset(MemoryCoreEvent event, RegisterOffset gep) {
+            assert gep == null || gep.register != null;
```

Comment:
Shouldn't this be `assert gep != null || gep.register != null;`?
If `gep` is null, then `gep.register` will throw an exception

Reply:
This is an implication, that allows `gep` to be null

Reply:
I can never get the re-write of an implication right when reading code :/